tests/integration/ha/helpers.py

```diff
@@ -0,0 +1,458 @@
+        ]
+    )
+
+
+def get_unit_host(
+    ops_test: OpsTest, unit_name: str, app_name: str = APP_NAME, port: int = 2181
+) -> str:
+    """Gets ZooKeeper server address for a given unit name.
+
+    Args:
+        ops_test: OpsTest
+        unit_name: the Juju unit to get host from
+        app_name: the Juju application the unit belongs to
+            Defaults to `zookeeper`
+        port: the desired ZooKeeper port.
+            Defaults to `2181`
+
+    Returns:
+        String of ZooKeeper server address and port
+    """
+    return [
+        f"{unit.public_address}:{str(port)}"
+        for unit in ops_test.model.applications[app_name].units
+        if unit.name == unit_name
+    ][0]
+
+
+def get_unit_name_from_host(ops_test: OpsTest, host: str, app_name: str = APP_NAME) -> str:
+    """Gets unit name for a given ZooKeeper server address.
+
+    Args:
+        ops_test: OpsTest
+        host: the ZooKeeper ip address and port
+        app_name: the Juju application the ZooKeeper server belongs to
+            Defaults to `zookeeper`
+
+    Returns:
+        String of unit name
+    """
+    return [
+        unit.name
+        for unit in ops_test.model.applications[app_name].units
+        if unit.public_address == host.split(":")[0]
+    ][0]
+
+
+def get_leader_name(ops_test: OpsTest, hosts: str, app_name: str = APP_NAME) -> str:
+    """Gets current ZooKeeper quorum leader for a given application.
+
+    Args:
+        ops_test: OpsTest
+        hosts: comma-delimited list of ZooKeeper ip addresses and ports
+        app_name: the Juju application the unit belongs to
+            Defaults to `zookeeper`
+
+    Returns:
+        String of unit name of the ZooKeeper quorum leader
+    """
+    for host in hosts.split(","):
+        host, port = host.split(":")
+        try:
+            mode = srvr(host, port)["Mode"]
+        except subprocess.CalledProcessError:  # unit is down
+            continue
+        if mode == "leader":
+            leader_name = get_unit_name_from_host(ops_test, host, app_name)
+            return leader_name
+
+    return ""
+
+
+async def get_unit_machine_name(ops_test: OpsTest, unit_name: str) -> str:
+    """Gets current LXD machine name for a given unit name.
+
+    Args:
+        ops_test: OpsTest
+        unit_name: the Juju unit name to get from
+
+    Returns:
+        String of LXD machine name
+            e.g juju-123456-0
+    """
+    _, raw_hostname, _ = await ops_test.juju("ssh", unit_name, "hostname")
+    return raw_hostname.strip()
+
+
+def disable_lxd_dnsmasq() -> None:
+    """Disables DNS resolution in LXD."""
+    disable_dnsmasq_cmd = "lxc network set lxdbr0 dns.mode=none"
+    subprocess.check_call(disable_dnsmasq_cmd.split())
+
+
+def enable_lxd_dnsmasq() -> None:
+    """Disables DNS resolution in LXD."""
+    enable_dnsmasq = "lxc network unset lxdbr0 dns.mode"
+    subprocess.check_call(enable_dnsmasq.split())
+
+
+def cut_unit_network(machine_name: str) -> None:
```

Comment:
`cut` is IP cahnge, `throttle` is no IP change.

I can add it in the docstring (I thought it was there though but I may have missed)

Reply:
I have taken a careful attention to call the actual test functions according to your requested naming (i.e. indicating in the test function name `ip_change`, `no_ip_change`.

Regarding helper functions, I would like to stick to the current naming -- while I made sure to very clearly indicate in the docstrings what is the impact on the IP address.

I may not be the only one to who can distinguish a lot easier between `cut` and `throttle` words than seeking for a small `no` tag within long function names. This is the rationale why I would I find the current naming handy, and I would like to stick to it. While assuring clarity (docstrings) and following your naming on the actual "interface" i.e. pytest executable test functions. 

I hope this could be a good compromise for both of us :-)